### PR TITLE
NC | Noobaa log is not performing logrotate 

### PR DESCRIPTION
**Pull request by naveenpaul1**
### Explain the changes
1. Enhance logrotate function to support syslog-ng if it used
2. logrotate will happen daily and logrotate-noobaa will be used for the rotation. Outchannel is removed from rsyslog config and rotation will not happen when log file reach size limit after this changes.
3. ratelimit for rsyslog is disabled
4. rsyslog and logrotate config files are unchanged by RPM, if edited : [link](https://www.cl.cam.ac.uk/~jw35/docs/rpm_config.html)

### Issues: Fixed #xxx / Gap #xxx
1. https://github.com/noobaa/noobaa-core/issues/8098
5. https://github.com/noobaa/noobaa-core/issues/8153

### Testing Instructions:
1. install noobaa RPM
2. check logs are getting rotated daily basis when logs are reachs configured size

- [X] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/deploy/RPM_build/RPM.Dockerfile [hunk 1]
```diff
@@ -36,7 +35,6 @@ COPY ./binding.gyp .
 COPY ./src/deploy/standalone/noobaa_rsyslog.conf ./src/deploy/standalone/noobaa_rsyslog.conf
 COPY ./src/deploy/standalone/noobaa_syslog.conf ./src/deploy/standalone/noobaa_syslog.conf
 COPY ./src/deploy/standalone/logrotate_noobaa.conf ./src/deploy/standalone/logrotate_noobaa.conf
-COPY ./src/deploy/standalone/logrotate_noobaa.sh ./src/deploy/standalone/logrotate_noobaa.sh
```
Comment (guymguym) [inline src/deploy/RPM_build/RPM.Dockerfile:39]: what about NooBaa.Dockerfile -
https://github.com/noobaa/noobaa-core/blob/1e26ec2ec13cb9a98455e613673283f8207bef52/src/deploy/NVA_build/NooBaa.Dockerfile#L113C51-L114C1

File: docs/non_containerized_NSFS.md [hunk 2]
```diff
@@ -582,14 +582,14 @@ systemctl status rsyslog
...
+Verify the rsyslog and logrotate rpm configuration is complete by checking the files `etc/rsyslog.d/noobaa_syslog.conf` for rsyslog and `etc/logrotate.d/logrotate_noobaa.conf` for logrotate.These files contain the noobaa specific configuration for rsyslog and logrotate.
 
-Logrotate configuration is set up under `/etc/logrotate.d/noobaa/`. In order to trigger rotation when log files reache size threshold, each log file (`/var/log/noobaa.log` and `/var/log/noobaa_events.log`) uses its own rsyslog `outchannel` that triggers logrotate once file size reaches a limit (105MB). Logrotate is also typically called from cron on daily schedule.
+Logrotate configuration is set up under `/etc/logrotate.d/`.  Logrotate is called from cron on daily schedule.
 
 To rotate the logs manually run.
 
 ```
-logrotate /etc/logrotate.d/noobaa/logrotate_noobaa.conf
+logrotate /etc/logrotate.d/logrotate_noobaa.conf
```
Comment (guymguym) [inline docs/non_containerized_NSFS.md:592]: what about this doc -
https://github.com/noobaa/noobaa-core/blob/1e26ec2ec13cb9a98455e613673283f8207bef52/src/deploy/spectrum_archive/deployment_guide.md?plain=1#L150-L167

File: src/deploy/RPM_build/RPM.Dockerfile (hunk 1 above)
Comment (guymguym) [reply src/deploy/RPM_build/RPM.Dockerfile:39]: I also see it in this test -
https://github.com/noobaa/noobaa-core/blob/1e26ec2ec13cb9a98455e613673283f8207bef52/src/test/unit_tests/test_debug_module.js#L93

Comment (naveenpaul1) [reply src/deploy/RPM_build/RPM.Dockerfile:39]: @guymguym these two files are not part of NC NSFS logrotate, This code was there before

Comment (guymguym) [reply src/deploy/RPM_build/RPM.Dockerfile:39]: ok, but is there any reason why it should be different there? It just adds confusion leaving some cases still use the extra noobaa dir...

File: src/deploy/spectrum_archive/deployment_guide.md [hunk 3]
```diff
@@ -157,12 +157,12 @@ systemctl status rsyslog
...
 Noobaa logs are pushed to `/var/log/noobaa.log` and the log is rotated and compressed daily.
 
-Verify the rsyslog and logrotate rpm configuration is complete by checking the files `/etc/rsyslog.d/noobaa_syslog.conf` and `/etc/rsyslog.d/noobaa_rsyslog.conf` for rsyslog and `/etc/logrotate.d/noobaa/logrotate_noobaa.conf` for logrotate.These files contain the noobaa specific configuration for rsyslog and logrotate.
+Verify the rsyslog and logrotate rpm configuration is complete by checking the files `/etc/rsyslog.d/noobaa_syslog.conf` and `/etc/rsyslog.d/noobaa_rsyslog.conf` for rsyslog and `/etc/logrotate.d/logrotate_noobaa.conf` for logrotate.These files contain the noobaa specific configuration for rsyslog and logrotate.
 
 Rotate the logs manually.
 
 ```
-logrotate /etc/logrotate.d/noobaa/logrotate_noobaa.conf
+logrotate /etc/logrotate.d/logrotate_noobaa.conf
```
Comment (guymguym) [inline src/deploy/spectrum_archive/deployment_guide.md:165]: OK one more request - I see in /etc/logrotate.d that we are not following the same pattern as other logrotate config files, and I would like to change our `src/deploy/standalone/logrotate_noobaa.conf` file to be named `src/deploy/standalone/noobaa-logrotate` (start with service name, and without .conf), and also deploy it by that same name.

**Review by guymguym — APPROVED** [review]

**Comment by aspandey** [comment]
Looks good to me. 
Just change the name of the file in first description comment from logrotate_noobaa.config to logrotate-noobaa

**Pull request merged**